src/Geometry/Primitives/AsphericSurface.jl

```diff
@@ -0,0 +1,341 @@
+
+The function checks if the aspheric terms are even, odd or both and uses `EvenAsphericSurface()`,
+`OddAsphericSurface()`, or `OddEvenAsphericSurface()` as appropriate.
+
+"""
+#"pseudo-types" of aspheres
+const CONIC = 0
+const ODD = 1
+const EVEN = 2
+const ODDEVEN = 3
+
+struct AsphericSurface{T,N, Q, M} <: ParametricSurface{T,N} 
+    semidiameter::T
+    curvature::T
+    conic::T 
+    aspherics::SVector{Q,T}   
+    normradius::T
+    boundingcylinder::Cylinder{T,N}
+
+    function AsphericSurface(semidiameter::T; radius::T = typemax(T), conic::T= zero(T), aspherics::Union{Nothing,Vector{Tuple{Int,T}}} = nothing, normradius::T = semidiameter) where {T<:Real}
+        @assert semidiameter > 0
+        @assert !isnan(semidiameter) && !isnan(radius) && !isnan(conic)
+        @assert one(T) - (1 / radius)^2 * (conic + one(T)) * semidiameter^2 > 0 "Invalid surface (conic/radius combination: $radius, $conic)"
+
+        acs = []
+        if aspherics===nothing
+            surf = new{T,3, 0, CONIC}(semidiameter, 1/radius, conic, Vector{T}(acs), normradius, Cylinder{T,3}(semidiameter, interface = opaqueinterface(T)))
+        else
+            asphericTerms = [i for (i, ) in aspherics]
+            minAsphericTerm = minimum(asphericTerms)
+            maxAsphericTerm = maximum(asphericTerms)
+            @assert minAsphericTerm > 0 "Aspheric Terms must be Order 1 or higher ($minAsphericTerm)"
+            acs = zeros(T, maxAsphericTerm )
+            for (i, k) in aspherics
+                acs[i] = k
+            end
+            odd = any([isodd(i) && a != zero(T) for (i, a) in enumerate(acs)])
+            even = any([iseven(i) && a != zero(T) for (i, a) in enumerate(acs)])
+            if odd && even
+                Q = maxAsphericTerm
+                surf = new{T, 3, Q, ODDEVEN}(semidiameter, 1 / radius, conic, Vector{T}(acs), normradius, Cylinder(semidiameter, interface = opaqueinterface(T))) 
+            elseif even
+                eacs = [acs[2i] for i in 1:(maxAsphericTerm ÷ 2)] 
+                Q = length(eacs)
+                surf = new{T, 3, Q, EVEN}(semidiameter, 1 / radius, conic, Vector{T}(eacs), normradius, Cylinder(semidiameter, interface = opaqueinterface(T))) 
+            else #odd
+                oacs = [acs[2i-1] for i in 1:((maxAsphericTerm+1) ÷ 2)]
+                Q = length(oacs)
+                surf = new{T, 3, Q, ODD}(semidiameter, 1 / radius, conic, Vector{T}(oacs), normradius, Cylinder(semidiameter, interface = opaqueinterface(T))) 
+            end
+        end
+        return surf
+    end
+    function EvenAsphericSurface(semidiameter::T, curvature::T, conic::T, aspherics::Vector{T}; normradius::T=semidiameter) where T<:Real
+        Q=length(aspherics)
+        new{T,3, Q, EVEN}(semidiameter, curvature, conic, SVector{Q,T}(aspherics), normradius, Cylinder(semidiameter, interface = opaqueinterface(T)))
```

Comment:
```suggestion
        new{T,3,Q,EVEN}(semidiameter, curvature, conic, SVector{Q,T}(aspherics), normradius, Cylinder(semidiameter, interface = opaqueinterface(T)))
```
Formmating (and elsewhere)